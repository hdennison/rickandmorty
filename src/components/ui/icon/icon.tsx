import type { LucideIcon, LucideProps } from "lucide-react";
import { Circle, ChevronsUpDown, Search } from "lucide-react";
import { cn } from "../utils";

type IconProps = LucideProps & { className?: string };

const withClassName = (Icon: LucideIcon, displayName: string) => {
  const WrappedIcon = ({ className, ...props }: IconProps) => (
    <Icon className={cn("shrink-0", className)} {...props} />
  );

  WrappedIcon.displayName = displayName;
  return WrappedIcon;
};

export const CircleIcon = withClassName(Circle, "CircleIcon");
export const SearchIcon = withClassName(Search, "SearchIcon");
export const ChevronsUpDownIcon = withClassName(
  ChevronsUpDown,
  "ChevronsUpDownIcon",
);
