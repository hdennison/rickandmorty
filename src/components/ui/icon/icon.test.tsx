import { expect, test } from "vitest";
import { render } from "vitest-browser-react";

import { ChevronsUpDownIcon, CircleIcon, SearchIcon } from "./icon";

const Icons = [
  { name: "ChevronsUpDownIcon", Icon: ChevronsUpDownIcon },
  { name: "CircleIcon", Icon: CircleIcon },
  { name: "SearchIcon", Icon: SearchIcon },
];

Icons.forEach(({ name, Icon }) => {
  test(name, () => {
    const icon = render(<Icon />);

    // Snapshots are not the best solution, it woould be much better to set up Storybook + Chromatic
    expect(icon).toMatchSnapshot();
  });
});

test("Icon size", () => {
  const icon = render(<CircleIcon size={16} />);

  // Snapshots are not the best solution, it woould be much better to set up Storybook + Chromatic
  expect(icon).toMatchSnapshot();
});
