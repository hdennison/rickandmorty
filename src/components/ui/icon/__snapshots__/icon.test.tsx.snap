// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChevronsUpDownIcon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <svg
        class="lucide lucide-chevrons-up-down shrink-0"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m7 15 5 5 5-5"
        />
        <path
          d="m7 9 5-5 5 5"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="lucide lucide-chevrons-up-down shrink-0"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m7 15 5 5 5-5"
      />
      <path
        d="m7 9 5-5 5 5"
      />
    </svg>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CircleIcon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <svg
        class="lucide lucide-circle shrink-0"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="lucide lucide-circle shrink-0"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Icon size 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <svg
        class="lucide lucide-circle shrink-0"
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="lucide lucide-circle shrink-0"
      fill="none"
      height="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchIcon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <svg
        class="lucide lucide-search shrink-0"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <path
          d="m21 21-4.3-4.3"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="lucide lucide-search shrink-0"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <path
        d="m21 21-4.3-4.3"
      />
    </svg>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
